code/go/0chain.net/blobbercore/handler/object_operation_grpc_handler.go

```diff
@@ -0,0 +1,114 @@
+package handler
+
+import (
+	"context"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/allocation"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/blobbergrpc"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/reference"
+	"github.com/0chain/blobber/code/go/0chain.net/core/common"
+	"github.com/0chain/blobber/code/go/0chain.net/core/lock"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+	"path/filepath"
+)
+
+func (b *blobberGRPCService) CopyObject(ctx context.Context, r *blobbergrpc.CopyObjectRequest) (
+	*blobbergrpc.CopyObjectResponse, error) {
+
+	logger := ctxzap.Extract(ctx)
+
+	allocationTx := r.Allocation
+	allocationObj, err := b.storageHandler.verifyAllocation(ctx, allocationTx, false)
+	if err != nil {
+		return nil, common.NewError("invalid_parameters", "Invalid allocation id passed."+err.Error())
+	}
+
+	md := GetGRPCMetaDataFromCtx(ctx)
+	valid, err := verifySignatureFromRequest(allocationTx, md.ClientSignature, allocationObj.OwnerPublicKey)
+	if !valid || err != nil {
+		return nil, common.NewError("invalid_signature", "Invalid signature")
+	}
+
+	clientID := md.Client
+	allocationID := allocationObj.ID
+
+	if len(clientID) == 0 {
+		return nil, common.NewError("invalid_operation", "Invalid client")
```

Comment:
Will read this file in next review but please make sure that the changes between this and original is minimal, thanks!